### PR TITLE
Add interactivity to the legend, add styles for percentage label

### PR DESCRIPTION
**Pull request by kullJul**

### CODE REVIEW

File: src/SunburstUtils.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+export class SunburstUtils {
+    static DimmedOpacity: number = 0.3;
+    static DefaultOpacity: number = 1.0;
+    static DimmedColor: string = "#A6A6A6";
+
+    static getOpacity(selected: boolean, highlight: boolean, hasSelection: boolean, isHCM: boolean): number {
+        if (!highlight && hasSelection && !selected && isHCM) {
```
[the remark this answers is not in the document]
Comment (kullJul) [reply src/SunburstUtils.ts]: fixed a210a6ae673b42308dce9dee6abb6e2460c77a77

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "powerbi-visuals-sunburst",
   "description": "SunBurst chart.",
-  "version": "2.5.1.0",
+  "version": "2.5.2.0",
```
[the remark this answers is not in the document]
Comment (kullJul) [reply package.json]: done a70149bef428f6fd10244019333aff4881bd9de7

File: src/Sunburst.ts [hunk 3]
```diff
@@ -664,15 +672,24 @@ export class Sunburst implements IVisual {
     }
 
     private setPercentageLabelPosition(width: number, canDisplayCategory: boolean): void {
-        const labelSize: number = this.settings.group.selectedCategory.font.fontSize.value * Sunburst.PercentageFontSizeMultiplier;
+        const percentageLabelSettings = this.settings.centralLabel.percentageLabel;
```
Comment (AleksSavelev) [inline src/Sunburst.ts:675]: I would suggest: `const fontSettings = this.settings.centralLabel.percentageLabel.font`
just for less code

File: src/SunburstSettings.ts [hunk 4]
```diff
@@ -88,21 +88,55 @@ class BaseFontCardSettings extends formattingSettings.FontControl {
...
         name: "showSelected",
         displayNameKey: "Visual_ShowCategoryLabel",
         value: this.defaultShowSelected,
     });
-    public font = new BaseFontCardSettings(this.defaultFontSize);
+
+    public indentation = new formattingSettings.Slider({
+        name: "indentation",
+        displayNameKey: "Visual_Indentation",
+        value: this.defaultIndentation
```
Comment (AleksSavelev) [inline src/SunburstSettings.ts:104]: Good to have limitations for all numeric settings.

File: src/SunburstSettings.ts [hunk 5]
```diff
@@ -88,21 +88,55 @@ class BaseFontCardSettings extends formattingSettings.FontControl {
...
+    });
+
+    public customizeStyle = new formattingSettings.ToggleSwitch({
+        name: "customizeStyle",
+        displayNameKey: "Visual_CustomizeStyle",
+        value: this.defaultCustomizeStyle,
+    });
+
+    public font = new BaseFontCardSettings(this.defaultFontSize, "categoryFont", "categoryFontFamily", "categoryFontSize", "categoryFontBold", "categoryFontItalic", "categoryFontUnderline");
```
Comment (AleksSavelev) [inline src/SunburstSettings.ts:113]: Let's rework this base class. We can simply pass unique keyword, such as "category", "label", etc and by default add to it all other strings ("category" + "Font" and same with others)

File: src/SunburstUtils.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
+export class SunburstUtils {
+    static DimmedOpacity: number = 0.3;
+    static DefaultOpacity: number = 1.0;
+    static DimmedColor: string = "#A6A6A6";
+
+    static getOpacity(selected: boolean, highlight: boolean, hasSelection: boolean, isHighContrast: boolean): number {
```
Comment (AleksSavelev) [inline src/SunburstUtils.ts:6]: Can't see it to be used

File: src/behavior.ts [hunk 7]
```diff
@@ -26,174 +26,193 @@
...
-                x: event.clientX,
-                y: event.clientY
+    private setSelectedToDataPoints(dataPoints: LegendDataPoint[] | HierarchyRectangularNode<SunburstDataPoint>[], ids: ISelectionId[]): void{
+        dataPoints.forEach((dataPoint: LegendDataPoint| HierarchyRectangularNode<SunburstDataPoint>) => {
+            const data : SunburstDataPoint | LegendDataPoint = (dataPoint as HierarchyRectangularNode<SunburstDataPoint>)?.data || (dataPoint as LegendDataPoint);
+            data.selected = false;
+            ids.forEach((selectedId: ISelectionId) => {
+                if (selectedId.includes(<ISelectionId>data.identity)) {
+                    data.selected = true;
+                }
```
Comment (AleksSavelev) [inline src/behavior.ts:90]: Not sure about the types here, but looks like it's possible to replace with 
```typescript
data.selected = ids.some((selectedId: ISelectionId) => selectedId.includes(<ISelectionId>data.identity))
```

File: src/behavior.ts [hunk 8]
```diff
@@ -26,174 +26,193 @@
...
-        });
-
-        clearCatcher.on("contextmenu", (event: PointerEvent) => {
-            this.selectionManager.showContextMenu({}, {
-                x: event.clientX,
-                y: event.clientY
-            });
+    private bindContextMenuEvent(elements: d3Selection<any>): void {
+        elements.on("contextmenu", (event: PointerEvent, dataPoint: HierarchyRectangularNode<SunburstDataPoint> | LegendDataPoint | undefined) => {
+            const data : SunburstDataPoint | LegendDataPoint = (dataPoint as HierarchyRectangularNode<SunburstDataPoint>)?.data || (dataPoint as LegendDataPoint);
```
Comment (AleksSavelev) [inline src/behavior.ts:97]: Looks complicated, what do  you think about moving it to a separate function (as it's used in several places)
Just like this
``` typescript
private castDataFromPoint(dataPoint: HierarchyRectangularNode<SunburstDataPoint> | LegendDataPoint | undefined) {
   return (dataPoint as HierarchyRectangularNode<SunburstDataPoint>)?.data || (dataPoint as LegendDataPoint);
}
```

**Review by Demonkratiy — APPROVED** [review]

**Pull request merged**